pkg/datastore/datastore.go

```diff
@@ -10,84 +10,140 @@ var (
 )
 
 func init() {
-	Cache = &memoryDatastore{
-		manifests: map[string]*OperatorManifest{},
+	Cache = newDataStore()
+}
+
+func newDataStore() *memoryDatastore {
+	return &memoryDatastore{
+		manifests: map[string]*operatorSourceRow{},
+		packages:  map[string]*SingleOperatorManifest{},
 		parser:    &manifestYAMLParser{},
+		walker:    &walker{},
+		bundler:   &bundler{},
 	}
 }
 
 // Reader is the interface that wraps the Read method.
 //
 // Read prepares an operator manifest for a given set of package(s)
 // uniquely represeneted by the package ID(s) specified in packageIDs. It
-// returns an instance of OperatorManifestData that has the required set of
+// returns an instance of RawOperatorManifestData that has the required set of
 // CRD(s), CSV(s) and package(s).
 //
 // The manifest returned can be used by the caller to create a ConfigMap object
 // for a catalog source (CatalogSource) in OLM.
 type Reader interface {
-	Read(packageIDs []string) (marshaled *OperatorManifestData, err error)
+	Read(packageIDs []string) (marshaled *RawOperatorManifestData, err error)
 }
 
 // Writer is an interface that is used to manage the underlying datastore
 // for operator manifest.
 type Writer interface {
-	// GetPackageIDs returns a comma separated list of IDs of
-	// all package(s) from underlying datastore.
+	// GetPackageIDs returns a comma separated list of operator ID(s). Each ID
+	// returned can be used to retrieve the manifest associated with the
+	// operator from underlying datastore.
 	GetPackageIDs() string
 
-	// Write stores the list of operator manifest(s) into datastore.s
-	Write(packages []*OperatorMetadata) error
+	// Write stores the list of operator manifest(s) into datastore.
+	Write(rawManifests []*OperatorMetadata) error
+}
+
+// operatorSourceRow is what gets stored in datastore after an OperatorSource CR
+// is reconciled.
+//
+// Every reconciled OperatorSource object has a corresponding operatorSourceRow
+// in datastore. The Writer interface accepts a raw operator manifest and
+// marshals it into this type before writing it to the underlying storage.
+type operatorSourceRow struct {
```

Comment:
That's a good question. `operatorSourceRow` is used internally by `datastore.go` is not part of any interface. 

I believe `SingleOperatorManifest` and `MultiOperatorManifest` are also internal to the package and don't need to be exported as well. The reason they were originally exported is because these types are part of interface(s). These types needed to be exported so that we can generate mocks (that lives outside of this package) for their respective interfaces.